Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/DocumentPropertyEditor/PrefabComponentAdapter.cpp

```diff
@@ -203,6 +150,57 @@ namespace AzToolsFramework::Prefab
             // out our reference to the undo batch here.
             m_currentUndoBatch = nullptr;
         }
+        AZ::Dom::Path serializedPath = propertyChangeInfo.path / AZ::Reflection::DescriptorAttributes::SerializedPath;
+
+        AZ::Dom::Path relativePathFromOwningPrefab(PrefabDomUtils::EntitiesName);
+        relativePathFromOwningPrefab /= m_entityAlias;
+        relativePathFromOwningPrefab /= PrefabDomUtils::ComponentsName;
+        relativePathFromOwningPrefab /= m_componentAlias;
+
+        AZ::Dom::Value serializedPathValue = GetContents()[serializedPath];
+        AZ_Assert(
+            serializedPathValue.IsString(), "PrefabComponentAdapter::UpdateDomContents - SerialziedPath attribute value is not a string.");
+
+        relativePathFromOwningPrefab /= AZ::Dom::Path(serializedPathValue.GetString());
+
+        auto prefabFocusPublicInterface = AZ::Interface<PrefabFocusPublicInterface>::Get();
+        if (prefabFocusPublicInterface->IsOwningPrefabBeingFocused(m_entityId))
+        {
+            // Normal component edit, so invoke the base behavior that updates the source template.
+            ComponentAdapter::UpdateDomContents(propertyChangeInfo);
+        }
+        else if (prefabFocusPublicInterface->IsOwningPrefabInFocusHierarchy(m_entityId))
+        {
+            // This is for an override, so in addition to the default replace operation,
+            // we need to also patch in the PrefabOverrideLabel in case the change doesn't
+            // trigger a refresh in the adapter.
+            AZ::Dom::Patch patches(
+                { AZ::Dom::PatchOperation::ReplaceOperation(propertyChangeInfo.path / "Value", propertyChangeInfo.newValue) });
+
+            AZ::Dom::Path pathToProperty = propertyChangeInfo.path;
+
+            // Get the path to parent row and its value.
+            pathToProperty.Pop();
+            AZ::Dom::Value propertyRowValue = GetContents()[pathToProperty];
+
+            AZ_Assert(
+                propertyRowValue.IsNode() && propertyRowValue.GetNodeName().GetStringView() == AZ::DocumentPropertyEditor::Nodes::Row::Name,
+                "PrefabComponentAdapter::UpdateDomContents - Parent path to property doesn't map to a 'Row' node. ");
+
+            AZ::Dom::Value firstRowElement = propertyRowValue[0];
+            AZ_Assert(
+                firstRowElement.IsNode() &&
+                    firstRowElement.GetNodeName().GetStringView() == AZ::DocumentPropertyEditor::Nodes::PropertyEditor::Name,
+                "PrefabComponentAdapter::UpdateDomContents - First element in the property row is not a PropertyEditor node.");
```

Comment:
Should there be a check that the `Dom::Value` is not [ArrayEmpty](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/DOM/DomValue.h#L305)? so that the `propertyRowValue[0]` doesn't index out of bounds?

Reply:
sorry, missed this, but I can address it in a future PR. That would only occur for a completely empty row, which shouldn't happen, but I can assert on it as a precaution, iyl